common/ferrostar/src/navigation_controller/mod.rs

```diff
@@ -366,6 +400,31 @@ impl NavigationController {
             _ => false,
         }
     }
+
+    /// Updates the trip summary with accumulated distance and duration.
+    fn update_trip_summary(
+        &self,
+        current_summary: &TripSummary,
+        distance_increment: u32,
+        snapped_distance_increment: u32,
+    ) -> TripSummary {
```

Comment:
This doesn't really need to be a method on `NavigationController`, since it is a pure function referencing only its arguments.

I think this might actually make more sense to have as a method on `TripSummary` directly, which consumes `self` (owned `self`; not a reference) and returns a new value. Something like:

`fn update(self,  previous_location: &UserLocation, current_location: &UserLocation) -> Self`